### PR TITLE
Nested folders: Create basic Move/Delete modals 

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- adds basic skeletons for the `Move`/`Delete` modals
  - these don't call any real apis yet, so they don't do anything meaningful. think of this PR as more of a review of the modal styling/design than anything 😄 
- adds basic unit tests
- changes the `ConfirmModal` e2e selector to use a `data-testid` instead of an `aria-label`
  - the `aria-label` was causing the unit tests to fail and is just an objectively worse label than the displayed text

**Why do we need this feature?**

- so users can move/delete dashboards when using nested folders

**Who is this feature for?**

- anyone using nested folders

**Which issue(s) does this PR fix?**:

For https://github.com/grafana/grafana/issues/65679
For https://github.com/grafana/grafana/issues/65681

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/LogGroups/SelectedLogGroups.test.tsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ describe('SelectedLogsGroups', () => {
       await waitFor(() =>
         expect(screen.getByText('Are you sure you want to clear all log groups?')).toBeInTheDocument()
       );
-      await waitFor(() => userEvent.click(screen.getByLabelText('Confirm Modal Danger Button')));
```
[the remark this answers is not in the document]
Comment (sarahzinger) [reply public/app/plugins/datasource/cloudwatch/components/LogGroups/SelectedLogGroups.test.tsx:71]: good point hadn't thought of that!

File: .betterer.results [hunk 2]
```diff
@@ -962,9 +962,6 @@ exports[`better eslint`] = {
       [0, 0, 0, "Unexpected any. Specify a different type.", "1"],
       [0, 0, 0, "Do not use any type assertions.", "2"]
     ],
-    "packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx:5381": [
-      [0, 0, 0, "Use data-testid for E2E selectors instead of aria-label", "0"]
-    ],
```
Comment (joshhunt) [inline .betterer.results:967]: nice.

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.test.tsx [hunk 3]
```diff
@@ -43,6 +42,6 @@ describe('ConfirmModal', () => {
     expect(screen.queryByText('Some Body')).not.toBeInTheDocument();
     expect(screen.queryByRole('button', { name: 'Dismiss Text' })).not.toBeInTheDocument();
     expect(screen.queryByRole('button', { name: 'Alternative Text' })).not.toBeInTheDocument();
-    expect(screen.queryByRole('button', { name: 'Confirm Modal Danger Button' })).not.toBeInTheDocument();
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.test.tsx:46]: Perfect example of how harmful old aria-label selectors are.

**Review by joshhunt — APPROVED** [review]
🚀

**Pull request merged**